src/history/test/HistoryTests.cpp

```diff
@@ -1264,3 +1264,142 @@ TEST_CASE("Catchup failure recovery with buffered checkpoint",
     // 3. Catchup to 191, and then externalize 194 to start catchup
     CHECK(catchupSimulation.catchupOnline(app, checkpointLedger, 1, 191, 3));
 }
+
+TEST_CASE("Change ordering of buffered ledgers after checkpoint",
+          "[history][catchup]")
+{
+    CatchupSimulation catchupSimulation{};
+
+    auto app = catchupSimulation.createCatchupApplication(
+        std::numeric_limits<uint32_t>::max(), Config::TESTDB_IN_MEMORY_SQLITE,
+        "app2");
+
+    auto checkpointLedger = catchupSimulation.getLastCheckpointLedger(1);
+    catchupSimulation.ensureOnlineCatchupPossible(checkpointLedger, 15);
+
+    // randomize externalize order of buffered ledgers
+    REQUIRE(
+        catchupSimulation.catchupOnline(app, checkpointLedger, 10, 0, 0, true));
```

Comment:
I think that instead of a randomize flag it would be better to pass in the actual order of ledgers in an optional vector (it also replaces the `ledgersToInject` vector) : that way you can easily test all the scenarios in this file with this parameter without introducing randomization (that actually doesn't really buy you anything here).
I'm thinking: generate range 1..128 (or whatever you need) and mutate it as you see it fit (for example, you can swap 64 and 65), you can add gaps, whatever you want there

Reply:
Done